From 5fd8f9a859392fa62d92841f4c2d65ac90b608a0 Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Thu, 21 May 2020 14:06:43 +0800
Subject: [PATCH] serial: 8250: fix compile warning by unused dma_err

drivers/tty/serial/8250/8250_port.c: In function 'serial8250_handle_irq':
drivers/tty/serial/8250/8250_port.c:1882:6: warning: unused variable 'dma_err' [-Wunused-variable]
  int dma_err = -1;
      ^~~~~~~

Fixes: 68ae37643a2b ("serial: 8250: support rx dma mode only")
Change-Id: I8c4ca9390777f7e8b07d5c6c329f005546c7d5fd
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 drivers/tty/serial/8250/8250_port.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/serial/8250/8250_port.c b/drivers/tty/serial/8250/8250_port.c
index ffa5a1df4ff8..064f64068362 100644
--- a/drivers/tty/serial/8250/8250_port.c
+++ b/drivers/tty/serial/8250/8250_port.c
@@ -1879,7 +1879,6 @@ int serial8250_handle_irq(struct uart_port *port, unsigned int iir)
 	unsigned char status;
 	unsigned long flags;
 	struct uart_8250_port *up = up_to_u8250p(port);
-	int dma_err = -1;
 
 	if (iir & UART_IIR_NO_INT)
 		return 0;
@@ -1890,6 +1889,8 @@ int serial8250_handle_irq(struct uart_port *port, unsigned int iir)
 
 #ifdef CONFIG_ARCH_ROCKCHIP
 	if (status & (UART_LSR_DR | UART_LSR_BI)) {
+		int dma_err = -1;
+
 		if (up->dma && up->dma->rxchan)
 			dma_err = handle_rx_dma(up, iir);
 
-- 
2.35.3

